larray_editor/arraymodel.py

```diff
@@ -353,11 +344,9 @@ def data(self, index, role=Qt.DisplayRole):
         elif role == Qt.BackgroundColorRole:
             if self.bgcolor_enabled and value is not np.ma.masked:
                 if self.bg_gradient is None:
-                    maxdiff = self.vmax - self.vmin
                     color_val = float(self.color_func(value))
-                    hue = self.hue0 + self.dhue * (self.vmax - color_val) / maxdiff
-                    color = QColor.fromHsvF(hue, self.sat, self.val, self.alp)
-                    return to_qvariant(color)
+                    bg_gradient = LinearGradient([(self.vmin, self.hsv_min), (self.vmax, self.hsv_max)])
```

Comment:
I may misunderstood the LinearGradient class. It is possible to change positions without changing colors, right?

Reply:
No, it is not really meant to work at the moment (*), but you can still only create the gradient object when vmin/vmax changes, not for each cell displayed.

Actually, setting the gradient .positions attribute directly should work, but I wouldn't do that unless we wrap that in a property to make sure .positions and .colors keep the same size.

If you go down that route, maybe changing the constructor to take positions and colors separately, would make it more obvious that we can change either individually.

Reply:
If we drop `bgcolor_enabled`, it means that we need to create a new LinearGradient when vmin and vman are changed but also in `bgcolor()` method if `state > 0`?

Reply:
Yes, I did not realize that. Forget it then, that seems more trouble that is worth.

Reply:
In `DataArrayModel.data()`, what is the purpose of the following piece of code? :
```python
    bg_value = self.bg_value
     x, y = index.row(), index.column()
     # FIXME: this is buggy on filtered data. We should change
     # bg_value when changing the filter.
     idx = y + x * bg_value.shape[-1]
     value = bg_value.data.flat[idx]
     return self.bg_gradient[value]
```
The background color of each cell is given by its position?

Reply:
the value used to compute the background color comes from another array. This is used to let compare color values depending on the difference between arrays instead of the array values themselves. This code is a hack though and (as the comment states) breaks as soon as the data is filtered.